src/test/java/org/apache/commons/text/similarity/JaccardDistanceTest.java

```diff
@@ -37,7 +37,7 @@ public static void setUp() {
     @Test
     public void testGettingJaccardDistance() {
         // Expected Jaccard distance = 1.0 - (intersect / union)
-        assertEquals(1.0, classBeingTested.apply("", ""));
+        assertEquals(0.0, classBeingTested.apply("", ""));
```

Comment:
This only works because the unit test passes in the same string on both sides, i.e. `"" vs ""`.

Do not use == to measure equality. This measures the same object reference. Try this in your test:

```
// Change 
assertEquals(0.0, classBeingTested.apply("", ""));
// to 
assertEquals(0.0, classBeingTested.apply("", new StringBuilder()));
```

You should fix the code to use `StringUtils.equals(CharSequence, CharSequence)` to test for equality.

Reply:
Thank you, but
1) Operator == is used to measure two int values (length).
2) with `new StringBuilder()` also works well

Reply:
Sorry, you are right. It was my mistake carried over from fixing another similarity class to use `StringUtils.equals`.

However you should consider using `StringUtils.equals` before splitting all the sequences into a set to provide a fast exit in the case of equal input. This is done in `JaroWinklerSimilarity`.